src/goto-programs/add_race_assertions.cpp

```diff
@@ -25,10 +26,13 @@ class w_guardst
 
     w_guards.push_back(identifier);
 
+    type2tc index = type2tc(array_type2tc(get_bool_type(), expr2tc(), true));
```

Comment:
Is the additional `type2tc()` around the array_type2tc needed? If so, it's a bug.